tests/Monolog/Formatter/JsonFormatterTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ErrorControl\Tests\Monolog\Formatter;
+
+use DateTimeImmutable;
+use Exception;
+use Keboola\ErrorControl\Monolog\Formatter\JsonFormatter;
+use Monolog\Level;
+use Monolog\LogRecord;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+
+class JsonFormatterTest extends TestCase
+{
+    public function testFormatting(): void
+    {
+        $logWithException = $this->givenLogWithException();
+        $formattedOutput = $this->whenLogIsFormatted($logWithException);
+        $this->thenOutputFormatIsCorrect($formattedOutput);
```

Comment:
Nemuzu si pomoc, ale i na me je tohle over-engineered :smile: Testuje to hrozne jednoduchou vec a vlastne z toho na prvni pohled nevidim co. Osobne bych to napsal spis jako

```
$logWithException = new LogRecord(
    new DateTimeImmutable(),
    'test',
    Level::Critical,
    'Message',
    [new Exception('Message')]
);

$formattedOutput = (new JsonFormatter())->format($logRecord);

$json = json_decode($output);
self::assertInstanceOf(stdClass, $json);

self::assertObjectHasProperty('trace', $json->context[0]);
self::assertIsString($json->context[0]->trace);
```

Porad je tam zachovany flow setup -> execute -> assert, ale nemusim lezt po funkcich, abych porovnal vstupy s vystupama. Tim nechci rict, ze je tabu v testech pouzivat funkce, spis jsme ale zvykli (snad muzu mluvit za vsechny :wink:) si delat nejaky helpery kdyz je treba slozitejsi setup a obsahuje nejakou nezajimavou omacku.

Reply:
Plus tady mas test jednoduchy exception, ktera tak nejak fungovala i pred tim. Bylo by dobry pridat jeste test na exception, ktera ma previous exception a ten assert upravit, at z jen je videt, jak by ten context mel spravne vypadat.

```
self::assertSame([
  'trace' => '...'
  'previous' => ...
], $json['context'])
```

Reply:
Ty cesty by mely byt lokalne i v CI stejne - bezi to oboje ve stejnym kontejneru. Ale ty cisla radku by mohly yt dost opruz, to mas pravdu, to mi nedocvaklo :+1: